app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt

```diff
@@ -40,6 +44,9 @@ class WorkoutProviderImpl : WorkoutProvider {
     private val exerciseHistory: Flow<List<ExerciseHistoryInfo>>
         get() = _exerciseHistory
 
+    private val _clickedPastWorkout = MutableStateFlow(HistoryInfoWorkout())
+    private val clickedPastWorkout: StateFlow<HistoryInfoWorkout> = _clickedPastWorkout
```

Comment:
We don't need this if it's private. We can have only one field. The second property is created to expose the flow as an immutable field to external classes.

Reply:
Update: see this comment: https://github.com/HauntedMilkshake/fitness_app/pull/46/files#r1865712693